### ISSUE TITLE
Implement --hls-prefer-livestreamer

### ISSUE THREAD

**Issue by Kagami** [rg3/youtube-dl#9349]
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2016.04.24**
- [x] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [x] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### Description of your *issue*, suggested solution and other information

FFmpeg seems to be very ineffecient on HLS streams, especially on a slow ones. For example, it drops a lot of segments on vlive.tv streams with middle/high quality (low-res streams are recorded fine). On the other hand `livestreamer "hls://stream-url.m3u8" best -o out.ts` works perfectly fine. So I think it would be great to have support for external livestreamer downloader. Note that `HlsFD` currently can't handle live streams.

(This is just a propose, I currently don't have any code for this.)

**Comment by yan12125** [comment]
Does livestreamer support other protocols? For example direct HTTP(S), DASH, MMS/RTSP, and RTMP?

**Comment by Kagami** [comment]
Yes, there are [few more](http://docs.livestreamer.io/cli.html#supported-streaming-protocols). Though RTMP seems to be handled via rtmpdump just like in youtube-dl.

**Comment by spookyahell** [comment]
I also think it would be great to be able to combine youtube-dl and livestreamer. For some videos I have to do it manually at this point, because ffmpeg will interrupt the download (even with "-timout" doesn't work good) and livestreamer will retry fetching and notify when it had trouble fetching a segment.

**Comment by Kagami** [comment]
livestreamer is de facto dead, but there is [streamlink fork](https://github.com/streamlink/streamlink) with same functionality.
